include/tesla.hpp

```diff
@@ -1814,21 +1915,44 @@ namespace tsl {
              * @brief Adds a new item to the list before the next frame starts
              * 
              * @param element Element to add
+             * @param index Index in the list where the item should be inserted. -1 or greater list size will insert it at the end
              * @param height Height of the element. Don't set this parameter for libtesla to try and figure out the size based on the type 
              */
-            virtual void addItem(Element *element, u16 height = 0) final {
+            virtual void addItem(Element *element, std::optional<size_t> index = {}, u16 height = 0) final {
```

Comment:
Why is this an std::optional? Couldn't it just be a size_t with default value -1?

Reply:
size_t is unsigned so smaller -1 is > 0

Reply:
`std::ssize_t` exists